src/lib_json/json_value.cpp

```diff
@@ -1018,6 +1018,34 @@ Value Value::removeMember(const std::string& key) {
   return removeMember(key.c_str());
 }
 
+bool Value::removeIndex(ArrayIndex index, Value* removed) {
+  if (this->type_ != arrayValue) {
+    return false;
+  }
+#ifdef JSON_VALUE_USE_INTERNAL_MAP
+  JSON_FAIL_MESSAGE("removeIndex is not implemented for ValueInternalArray.");
+  return false;
+#else
+  CZString key(index);
+  ObjectValues::iterator it = this->value_.map_->find(key);
+  if (it == this->value_.map_->end()) {
+    return false;
+  }
+  *removed = it->second;
+  ArrayIndex oldSize = this->size();
+  // shift left all items left, into the place of the "removed"
+  for (ArrayIndex i=index; i<oldSize-1; i++){
```

Comment:
Ironically, the prefix ++i notation is more idiomatic in C++, isn't it?
We might want a little whitespace in this expression. Other initializers nearby have spaces around the =, for example.